### ISSUE TITLE
btree iterator doesn't appear to work

### ISSUE THREAD

**Issue by freekvw** [thomasmoelhave/tpie#216]
The increment operation on the iterator in tpie::btree does not always terminate.

**Comment by Tyilo** [comment]
Is this still a problem?

If so, do you have a test case?

**Comment by freekvw** [comment]
I just checked, and currently we crash when the iterator appears to get to the end.  Jakob is working on a testcase, but isn't done yet.

**Comment by antialize** [comment]
Running
tpie/build/apps/siv_dump/siv_dump /tmp/dataset.siv
faults while
tpie/build/apps/siv_dump/siv_dump --recursive /tmp/dataset.siv
succedes

[example.zip](https://github.com/thomasmoelhave/tpie/files/1889056/example.zip)

**Comment by Tyilo** [comment]
I found the problem: For serialized btrees `t.end() != t.end()` (same with `begin`). I'm working on a fix.

**Comment by antialize** [comment]
I don't think this fix works.

if i do

for (auto it=tree.begin(); it != tree.end(); ++it) {...}

It appears to fail as a new end node is constructed and deleted for every loop iteration. Also these lookups in unordered maps could be slow. Would it not be better to fix the comparison to use the offset?

**Comment by Tyilo** [comment]
Works on my computer.

**Comment by antialize** [comment]
We should not use unordred maps to cache nodes. We should fix the comparision..

**Comment by Tyilo** [comment]
Fair enough, should be fixed in 1b6534f632c38d8cf1bf62cf3065c9378d97a192

**Comment by SSoelvsten** [comment]
Is this still a problem?
